lib/internal/test_runner/test.js

```diff
@@ -517,12 +524,11 @@ class Test extends AsyncResource {
         return;
       }
 
-      if (this.parent?.hooks.afterEach.length > 0) {
-        await this.parent[kRunHook]('afterEach', { args, ctx });
-      }
-
+      await afterEach();
       this.pass();
     } catch (err) {
+      // eslint-disable-next-line no-empty
+      try { await afterEach(); } catch {}
```

Comment:
FYI you can avoid the `eslint-disable` if you put a comment in the empty block, e.g.
```suggestion
      try { await afterEach(); } catch { /* test is already failing, let's the error */ }
```